tests/components/switch_as_x/test_light.py

```diff
@@ -0,0 +1,174 @@
+    assert hass.states.get("light.decorative_lights").state == "off"
+
+
+async def test_switch_service_calls(hass):
+    """Test service calls to switch."""
+    await async_setup_component(hass, "switch", {"switch": [{"platform": "demo"}]})
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": "switch.decorative_lights", "target_domain": "light"},
+        title="decorative_lights",
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("light.decorative_lights").state == "on"
+
+    await switch_common.async_turn_off(hass, "switch.decorative_lights")
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.decorative_lights").state == "off"
+    assert hass.states.get("light.decorative_lights").state == "off"
+
+    await switch_common.async_turn_on(hass, "switch.decorative_lights")
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.decorative_lights").state == "on"
+    assert hass.states.get("light.decorative_lights").state == "on"
+
+
+@pytest.mark.parametrize("target_domain", ("light",))
+async def test_config_entry(hass: HomeAssistant, target_domain):
+    """Test light switch setup from config entry."""
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": "switch.abc", "target_domain": target_domain},
+        title="ABC",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert DOMAIN in hass.config.components
+
+    state = hass.states.get(f"{target_domain}.abc")
+    assert state.state == "unavailable"
+    # Name copied from config entry title
+    assert state.name == "ABC"
+
+    # Check the light is added to the entity registry
+    registry = er.async_get(hass)
+    entity_entry = registry.async_get(f"{target_domain}.abc")
+    assert entity_entry.unique_id == config_entry.entry_id
+
+
+@pytest.mark.parametrize("target_domain", ("light",))
+async def test_config_entry_uuid(hass: HomeAssistant, target_domain):
+    """Test light switch setup from config entry with entity registry id."""
+    registry = er.async_get(hass)
+    registry_entry = registry.async_get_or_create("switch", "test", "unique")
+
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={"entity_id": registry_entry.id, "target_domain": target_domain},
+        title="ABC",
+    )
+
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.states.get(f"{target_domain}.abc")
+
+
+async def test_config_entry_unregistered_uuid(hass: HomeAssistant):
```

Comment:
Maybe move this to a test_init.py module? The code that we're testing is in the init module, I think. Not a big deal.

Reply:
Addressed in https://github.com/home-assistant/core/pull/67951